avidemux_plugins/ADM_videoEncoder/x265/qt4/Q_x265.cpp

```diff
@@ -79,28 +79,22 @@ static const aspectRatio predefinedARs[]={
 
 #define NB_SAR sizeof(predefinedARs)/sizeof(aspectRatio)
 
-typedef struct
-{
-    int bdValue;
-    const char *bdString;
-}bdToken;
-
-static const bdToken listOfBitDepths[]={
+static const idcToken listOfBitDepths[]={
     {8,"8"},
     {10,"10"},
     {12,"12"}
 };
 
-#define NB_BITS sizeof(listOfBitDepths)/sizeof(bdToken)
+#define NB_BITS sizeof(listOfBitDepths)/sizeof(*listOfBitDepths)
 
 static const char* listOfPresets[] = { "ultrafast", "superfast", "veryfast", "faster", "fast", "medium", "slow", "slower", "veryslow", "placebo" };
 #define NB_PRESET sizeof(listOfPresets)/sizeof(char*)
 
+static const char* listOfTunings[] = { "psnr", "ssim", "grain", "zerolatency", "fastdecode",
 #if X265_BUILD > 173
-static const char* listOfTunings[] = { "none", "psnr", "ssim", "grain", "zerolatency", "fastdecode", "animation" };
-#else
-static const char* listOfTunings[] = { "none", "psnr", "ssim", "grain", "zerolatency", "fastdecode" };
+    "animation"
```

Comment:
Are you sure this will work as expected with VC++? I had to do 1a3ab8a1f27f94ddbbc9f7556863f9bff8f56bdd in a similar situation.

Official Avidemux builds for Windows are built with Visual Studio, I haven't mastered the laborious setup of the build environment yet.

But this can be trivially changed later, after the PR is merged as-is.

Reply:
It should work.  It's a pretty common thing and totally within the standard.

I think the issue you see in 1a3ab8a is that it the ifdef is inside the invocation of a preprocessor macro.  That's not allowed.  The preprocessor is a simple one pass processor, so it isn't meant to processor the ifdefs inside a macro invocation, then expand the macro.  See https://stackoverflow.com/questions/39534609/is-it-wrong-to-add-preprocessor-directives-in-a-function-like-macro

This is just a variable initialization in plain C, so there is nothing here that will break VC++.

Reply:
> I think the issue you see in 1a3ab8a is that it the ifdef is inside the invocation of a preprocessor macro.

AFAIR, I had to abandon preprocessor macros in argument lists of functions like `void GUI_Error_HIG(const char *primary, const char *secondary_format, ...);` and in definition of `diaMenuEntry streamOpt[]` due to broken functionality, but maybe I misattributed the problem.

Thank you very much for the patch and excellent documentation.